docs/azrepos-users-and-tokens.md

```diff
@@ -0,0 +1,210 @@
+# Azure Repos: Access Tokens and Accounts
+
+## Different Credential Types
+
+The Azure Repos host provider supports creating multiple types of credential:
+
+- Personal Access Token
+- Azure Access Token (experimental)
+
+### Personal Access Tokens
+
+Historically, the only option supported by the Azure Repos host provider was
+Azure DevOps Personal Access Tokens (PATs).
+
+These PATs are only used by Azure DevOps, and must be [managed through the Azure
+DevOps user settings page](https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=preview-page).
+
+PATs have a limited lifetime and new tokens must be created once they expire. In
+Git Credential Manager, when a PAT expired (or was manually revoked) this
+resulted in a new authentication prompt.
+
+### Azure Access Tokens (experimental)
+
+Azure Access Tokens is the generic term for tokens issued by Azure Active
+Directory for either Work and School Accounts (AAD tokens) or Personal Accounts
+(Microsoft Account/MSA tokens).
+
+Azure DevOps supports Git authentication using Azure Access Tokens as well as
+PATs. Azure Access Tokens created by Git Credential Manager are scoped to Azure
+DevOps only.
+
+Unlike PATs, Azure Access Tokens get automatically refreshed and renewed as long
+as you are actively using them to perform Git operations.
```

Comment:
Am I reading this right that if we enable background maintenance, then the hourly prefetch action will keep the token alive indefinitely?

Reply:
AAD has access tokens and refresh tokens. It could either be helpful or confusing to call out the difference here - I'm not sure :)

https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-configurable-token-lifetimes#token-lifetime-policies-for-access-saml-and-id-tokens

Reply:
The document is simplifying somewhat (on purpose). The access tokens only last for an hour (typically) and are continually recreated from the longer-lived refresh token. The refresh token is _also_ renewed automagically by MSAL + ESTS as long as you're using the RT to mint a new AT. As the RT is getting close to expiry, ESTS will return a new RT along with the new AT.

> Am I reading this right that if we enable background maintenance, then the hourly prefetch action will keep the token alive indefinitely?

So yes, as long as we're using the ATs (and therefore using the RT) periodically, we'll never have an expired token.

The only cases where tokens would "expire" would be if the tenant administrator decided to enforce a conditional access policy to limit RT renewal lifetime, or make it conditional on something like being Intune/AAD joined.